### ISSUE TITLE
max doesn't get updated if first set to 0

### ISSUE THREAD

**Issue by x-yuri** [brownieboy/react-bootstrap-slider#50]
That is, if the first time the component has been mounted with `max={0}`, later changes doesn't take effect. This can be inferred from this block of [code][set-max]:

[set-max]: https://github.com/brownieboy/react-bootstrap-slider/blob/378d7b78425152150ad9c10f5660cc0b6c6905e0/src/js/react-bootstrap-slider.jsx#L99-L104

```js
        if (
            this.props.max &&
            (this.mySlider.max || this.mySlider.options.max)
        ) {
            this.mySlider.setAttribute("max", this.props.max);
        }
```

From what I can gather, `this.mySlider.max` is never set to anything. The only place for `max` to be set to `0` is [`componentDidMount`][cdm]. And if that happens, `max` can't be changed anymore.

But if the first value is `null`, it [works out][bs].

[bs]: https://github.com/seiyria/bootstrap-slider/blob/v10.0.0/src/js/bootstrap-slider.js#L381

[cdm]: https://github.com/brownieboy/react-bootstrap-slider/blob/378d7b78425152150ad9c10f5660cc0b6c6905e0/src/js/react-bootstrap-slider.jsx#L58

Judging from [commit][c1] [names][c2] it has something to do with ticks, but I don't see the connection.

[c1]: https://github.com/brownieboy/react-bootstrap-slider/commit/70820498e61f5079c2bd9b1ca31367e766623cb7
[c2]: https://github.com/brownieboy/react-bootstrap-slider/commit/00a82ce916ec0d15a6453c2b901f593569f0aba9

**Comment by brownieboy** [comment]
It might be because in JavaScript, zero is a representation of `false`.  Type `false == 0` in a browser and it should return `true`.

What I really should have done there is test that `max` is defined, like so:
```
if (
   typeof this.props.max !== "undefined"  &&
```
Probably on the other tests too.  I'll look into it.

**Comment by brownieboy** [comment]
Fixed and rolled out to npmjs as version 2.1.5.

Thanks for reporting.
